WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -78,14 +90,126 @@ public void onResume() {
     }
 
     @Override
+    protected void onPause() {
+        super.onPause();
+        if (mSearchMenuItem != null) {
+            String tempQuery = mQuery;
+            MenuItemCompat.collapseActionView(mSearchMenuItem);
+            mQuery = tempQuery;
+        }
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.posts_list, menu);
+
+        mSearchMenuItem = menu.findItem(R.id.search_posts_list);
+        MenuItemCompat.setOnActionExpandListener(mSearchMenuItem, this);
+
+        mSearchView = (SearchView) mSearchMenuItem.getActionView();
+        mSearchView.setOnQueryTextListener(this);
+
+        // open search bar if we were searching for something before
+        if (!TextUtils.isEmpty(mQuery)) {
+            String tempQuery = mQuery; //temporary hold onto query
+            MenuItemCompat.expandActionView(mSearchMenuItem); //this will reset mQuery
+            onQueryTextSubmit(tempQuery);
+            mSearchView.setQuery(mQuery, true);
+        }
+
+        return super.onCreateOptionsMenu(menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            case R.id.search_posts_list:
+                mSearchMenuItem = item;
+                mSearchView = (SearchView) item.getActionView();
+                mSearchView.setOnQueryTextListener(this);
+
+                if (!NetworkUtils.checkConnection(this)) {
+                    mSearchView.clearFocus();
+                    MenuItemCompat.collapseActionView(mSearchMenuItem);
+                }
+                else {
+                    MenuItemCompat.setOnActionExpandListener(mSearchMenuItem, this);
+                    MenuItemCompat.expandActionView(mSearchMenuItem);
+                    // load last saved query
+                    if (!TextUtils.isEmpty(mQuery)) {
+                        onQueryTextSubmit(mQuery);
+                        mSearchView.setQuery(mQuery, true);
+                    }
+                }
+
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public boolean onMenuItemActionExpand(MenuItem item) {
+        // load last search query
+        if (!TextUtils.isEmpty(mQuery)) {
+            onQueryTextChange(mQuery);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean onMenuItemActionCollapse(MenuItem item) {
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
```

Comment:
I thought if was needed because of how we're using `onQueryTextSubmit` when restoring state, but it works fine without explicitly searching here. Good eye!